frontend/packages/console-dynamic-plugin-sdk/src/utils/fetch/console-fetch.ts

```diff
@@ -42,7 +45,21 @@ const consoleFetchCommon = async (
   if (!text) {
     return isPlainText ? '' : {};
   }
-  return isPlainText || !response.ok ? text : JSON.parse(text);
+
+  // Map cosole respose to data, headers and status
+  const fullResponse: { data: Body; headers: { [key: string]: any }; status: number } = {
+    headers: [...response.headers].reduce((acc, header) => {
+      return { ...acc, [header[0]]: header[1] };
+    }, {}),
+    status: response.status,
+    data: JSON.parse(text),
+  };
+
+  return isPlainText || !response.ok
```

Comment:
We should still return the full response when set even if the mime type is `text/plain`.

Reply:
> We should still return the full response when set even if the mime type is `text/plain`.

@spadgett Not too sure of what you meant here because the `isPlainText` is returned in the current code. But I have removed the `isPlainText` in line 58 and returned full/raw response in line 46. PTAL, thanks.